extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java

```diff
@@ -1285,12 +1285,10 @@ public void verifyResults()
                                                                       .filter(segmentId -> segmentId.getInterval()
                                                                                                     .contains((Long) row[0]))
                                                                       .collect(Collectors.toList());
-            if (diskSegmentList.size() != 1) {
-              throw new IllegalStateException("Single key in multiple partitions");
-            }
-            SegmentId diskSegment = diskSegmentList.get(0);
             // Checking if the row belongs to the correct segment interval
-            Assert.assertTrue(segmentIdVsOutputRowsMap.get(diskSegment).contains(Arrays.asList(row)));
+            Assert.assertTrue(diskSegmentList.stream()
+                                             .map(segmentIdVsOutputRowsMap::get)
+                                             .anyMatch(rows -> rows.contains(Arrays.asList(row))));
```

Comment:
1. Each row is present in exactly one interval
2. Each row is present in exactly one segment. 


This is the behavior we want.

Reply:
Added an additional filter here to check that it only matches a certain segment and interval.